js/aside.js

```diff
@@ -1,4 +1,101 @@
-export const renderAside = (openModal) => {
-    document.querySelector(".add-event-button").addEventListener("click", openModal);
+import { isToday, areTwoDatesEqual } from './utils.js';
+
+export const renderAside = (currentDate, updateDate, openModal) => {
+  const currentMonth = document.getElementById("currentDateCalendarWidget");
+  const prevMonthButton = document.getElementById("prevMonth");
+  const nextMonthButton = document.getElementById("nextMonth");
+  const calendarDates = document.getElementById("calendarDates");
+  const addEventButton = document.getElementById("addEventButton");
+
+  const TOTAL_CELLS = 42;
+  let displayedMonth = new Date(currentDate.getFullYear(), currentDate.getMonth(), 1);
+  let selectedDate = new Date(currentDate);
+  const updateMonthDisplay = () => {
+    currentMonth.textContent = displayedMonth.toLocaleDateString("en-US", { year: 'numeric', month: 'short' });
+    renderMonthDays();
   };
-  
\ No newline at end of file
+
+  const createDateCell = (day, className, clickHandler) => {
+    const dateCell = document.createElement('div');
+    dateCell.textContent = day;
+    if (className) {
+      dateCell.classList.add(className);
+    }
+    if (clickHandler) {
+      dateCell.addEventListener('click', clickHandler);
+    }
+    calendarDates.appendChild(dateCell);
+    return dateCell;
+  };
+
+  const renderPreviousMonthDays = (startDay, previousMonthTotalDays) => {
+    for (let i = 0; i < startDay - 1; i++) { 
+      const day = previousMonthTotalDays - startDay + 2 + i;
+      const cellDate = new Date(displayedMonth.getFullYear(), displayedMonth.getMonth() - 1, day);
+      createDateCell(day, 'prev-month-day', () => {
+        displayedMonth.setMonth(displayedMonth.getMonth() - 1);
+        displayedMonth.setDate(day);
+        selectedDate = cellDate;
+        updateDate(selectedDate);
+        updateMonthDisplay();
+      });
+    }
+  };
+
+  const renderCurrentMonthDays = (totalDays) => {
+    for (let day = 1; day <= totalDays; day++) {
+      const cellDate = new Date(displayedMonth.getFullYear(), displayedMonth.getMonth(), day);
+      const className = isToday(cellDate) ? 'today' : (selectedDate && areTwoDatesEqual(cellDate, selectedDate)) ? 'selected' : '';
```

Comment:
nested ternary operators are not the best thing. two levels are still fine, but better should be avoided. and never more than two